src/modules/utils.js

```diff
@@ -8,6 +10,17 @@ const emptyLogger = {
   log: empty,
 };
 
+function setCorrelationId(options) {
+  if (options.correlationId) {
+    return options.correlationId;
+  }
+  const corrId = uuid.v4();
+  const updatedOptions = options;
+  updatedOptions.correlationId = corrId;
```

Comment:
```suggestion
  options.correlationId = corrId;
```

why we need the extra variable here?

Reply:
There is a lint rule forbidding directly mutating function arguments, so you need to set a new reference to it first.

Reply:
Changed it to just return the correlation id to avoid this.